arwen/host/execution.go

```diff
@@ -200,6 +226,11 @@ func (host *vmHost) finishExecuteOnDestContext(executeErr error) *vmcommon.VMOut
 
 	if vmOutput.ReturnCode == vmcommon.Ok {
 		output.PopMergeActiveState()
+		scAddress := string(runtime.GetSCAddress())
```

Comment:
Not sure, but - can we extract this and the same code from `finishExecuteOnSameContext` to a separate function `accumulateGasUsedByLibraryContract(vmOutput, scAddress, gasUsed)`? (optional, would be a bit more explicit)

Reply:
done